lib/rules/id-length.js

```diff
@@ -28,7 +30,7 @@ module.exports = function(context) {
                 // regular property assignment
                 parent.parent.left === parent || (
                     // or the last identifier in an ObjectPattern destructuring
-                    parent.parent.type === "Property" && parent.parent.value === parent &&
+                    parent.parent.type === "Property" && properties && parent.parent.value === parent &&
```

Comment:
If I check it before, wouldn't it mean the following would be valid (even though it shouldn't)? 

```js
const {abc: a} = {abc: 123};
```

There would be a constant `a` with value `123`, and that should be an error.